### ISSUE TITLE
v0.4.1 invalid memory address or nil pointer dereference crash loop

### ISSUE THREAD

**Issue by lanmarti** [argoproj-labs/argocd-notifications#27]
When trying to install v0.4.1 the argocd notifications pod enters a crash loop with the following output:
```
$ kubectl -n argocd logs argocd-notifications-controller-846b88966b-gkhfd
E0203 07:32:03.663598       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 35 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x141d520, 0x24bd4e0)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/runtime/runtime.go:48 +0x82
panic(0x141d520, 0x24bd4e0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
main.(*config).merge(0x24d9920, 0x0, 0x0)
        /src/cmd/main.go:228 +0x1ef
main.parseConfig(0x0, 0x0, 0x0, 0x0, 0x24f8620, 0xc0000865f8, 0x170d, 0x3, 0x0)
        /src/cmd/main.go:135 +0x66
main.watchConfig.func1(0x0, 0xc0003c2640)
        /src/cmd/main.go:163 +0x162
main.watchConfig.func2(0x1596b00, 0xc0003c2640)
        /src/cmd/main.go:175 +0x55
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
        /go/pkg/mod/k8s.io/client-go@v0.0.0-20191114101535-6c5935290e33/tools/cache/controller.go:198     k8s.io/client-go/tools/cache.(*processorListener).run.func1.1(0xc000001080, 0xc000198550, 0x0)
        /go/pkg/mod/k8s.io/client-go@v0.0.0-20191114101535-6c5935290e33/tools/cache/shared_informer.go:658 +0x26d
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc00006be38, 0x42ad4f, 0xc000198580)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:292 +0x51k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /go/pkg/mod/k8s.io/client-go@v0.0.0-20191114101535-6c5935290e33/tools/cache/shared_informer.go:652 +0x79
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00005c768)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:152 +0x54k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00006bf68, 0xdf8475800, 0x0, 0x42b501, 0xc000368780)             /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:153 +0xf8k8s.io/apimachinery/pkg/util/wait.Until(...)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:88       k8s.io/client-go/tools/cache.(*processorListener).run(0xc000344780)
        /go/pkg/mod/k8s.io/client-go@v0.0.0-20191114101535-6c5935290e33/tools/cache/shared_informer.go:650 +0x9c
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0003094f0, 0xc00003a6e0)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:71 +0x4f created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:69 +0x62 panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x12e530f]

goroutine 35 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/runtime/runtime.go:55 +0x105
panic(0x141d520, 0x24bd4e0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
main.(*config).merge(0x24d9920, 0x0, 0x0)
        /src/cmd/main.go:228 +0x1ef
main.parseConfig(0x0, 0x0, 0x0, 0x0, 0x24f8620, 0xc0000865f8, 0x170d, 0x3, 0x0)
        /src/cmd/main.go:135 +0x66
main.watchConfig.func1(0x0, 0xc0003c2640)
        /src/cmd/main.go:163 +0x162
main.watchConfig.func2(0x1596b00, 0xc0003c2640)
        /src/cmd/main.go:175 +0x55
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
        /go/pkg/mod/k8s.io/client-go@v0.0.0-20191114101535-6c5935290e33/tools/cache/controller.go:198     k8s.io/client-go/tools/cache.(*processorListener).run.func1.1(0xc000001080, 0xc000198550, 0x0)
        /go/pkg/mod/k8s.io/client-go@v0.0.0-20191114101535-6c5935290e33/tools/cache/shared_informer.go:658 +0x26d
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0x0, 0xc000449e38, 0x42ad4f, 0xc000198580)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:292 +0x51k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /go/pkg/mod/k8s.io/client-go@v0.0.0-20191114101535-6c5935290e33/tools/cache/shared_informer.go:652 +0x79
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00005c768)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:152 +0x54k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000449f68, 0xdf8475800, 0x0, 0x42b501, 0xc000368780)             /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:153 +0xf8k8s.io/apimachinery/pkg/util/wait.Until(...)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:88       k8s.io/client-go/tools/cache.(*processorListener).run(0xc000344780)
        /go/pkg/mod/k8s.io/client-go@v0.0.0-20191114101535-6c5935290e33/tools/cache/shared_informer.go:650 +0x9c
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0003094f0, 0xc00003a6e0)
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:71 +0x4f created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20191028221656-72ed19daf4bb/pkg/util/wait/wait.go:69 +0x62
```

For the first install, I updated directly from v0.3.0. When I noticed the crash loop, I removed everything ( ```kubectl delete -n argocd -f https://raw.githubusercontent.com/argoproj-labs/argocd-notifications/v0.4.1/manifests/install.yaml``` and ```kubectl delete -n argocd -f https://raw.githubusercontent.com/argoproj-labs/argocd-notifications/v0.3.0/manifests/install.yaml```) and tried v0.4.1 again, which still failed. v0.4.0 seems to be working.

Kubernetes version:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:23:11Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:13:49Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by alexmt** [comment]
Thanks a lot for bug report @lanmarti !

**Comment by alexmt** [comment]
published v0.4.2 with the fix: https://github.com/argoproj-labs/argocd-notifications/releases/tag/v0.4.2
